core/services/ocr2/delegate.go

```diff
@@ -643,16 +623,66 @@ func (d *Delegate) newServicesGenericPlugin(
 	pr := generic.NewPipelineRunnerAdapter(pluginLggr, jb, d.pipelineRunner)
 	ta := generic.NewTelemetryAdapter(d.monitoringEndpointGen)
 
-	plugin := reportingplugins.NewLOOPPService(pluginLggr, grpcOpts, cmdFn, pluginConfig, providerClientConn, pr, ta, errorLog)
-	oracleArgs.ReportingPluginFactory = plugin
-	srvs = append(srvs, plugin)
+	oracleEndpoint := d.monitoringEndpointGen.GenMonitoringEndpoint(
+		rid.Network,
+		rid.ChainID,
+		spec.ContractID,
+		synchronization.TelemetryType(p.TelemetryType),
+	)
 
-	oracle, err := libocr2.NewOracle(oracleArgs)
-	if err != nil {
-		return nil, err
+	if p.OCRVersion == 3 {
```

Comment:
Can we use a switch and explicitly check both versions?

That way if version == 1, we would error